custom_metrics/wpt_bodies.js

```diff
@@ -0,0 +1,1366 @@
+
+          let sameAsNodes = [...d.querySelectorAll('[itemprop="sameAs"]')];
+
+          sameAsNodes.forEach((n) => {
+            let href = n.getAttribute('href'); 
+
+            if (href) {
+              if (!target.same_as_values.includes(href)) {
+                target.same_as_values.push(href);
+              }
+            }
+          });
+
+          // rdfa
+          target.items_by_format.rdfa = d.querySelectorAll('[typeof]').length;
+
+          // microformats
+          target.microformats2_types = [];
+          ["h-adr","h-card","h-entry","h-event","h-feed","h-geo","h-item","h-listing draft","h-product","h-recipe","h-resume","h-review","h-review-aggregate"].forEach(name => {
+            let items = d.querySelectorAll('.'+name);
+            if (items.length > 0) {
+              target.microformats2_types.push({name: name, count: items.length});
+              target.items_by_format.microformats2 += items.length;
+            }
+          });
+
+          return target;
+        }
+
+        let result = {};
+        result.rendered = gatherStructuredData(document);
+
+        let rawHtmlDocument = getRawHtmlDocument();
+
+        if (rawHtmlDocument) {
+          result.raw = gatherStructuredData(rawHtmlDocument);
+        }
+        
+        return result; 
+      }
+      catch(e) {
+        return logError("structured_data", e);
+      }
+    })(),
+
+    // data from the original html
+    // Used by: SEO, Markup  
+    // Backup: queiries on the body - expensive
+    'raw_html': (() => {  
+      try {
+        let result = {};
+
+        let rawHtml = getRawHtml();
+
+        if (rawHtml) {
+
+          result.body = !!rawHtml.match(/<body/g);
+          result.html = !!rawHtml.match(/<html/g);
+          result.head = !!rawHtml.match(/<head/g);
+          result.size = rawHtml.length;
+
+          let headmatch = rawHtml.match(/<head.*<\/head>/gsi); // s = match newlines
+
+          if (headmatch) {
+            result.head_size = headmatch[0].length;
+          }
+
+          let bodymatch = rawHtml.match(/<body.*<\/body>/gsi); // s = match newlines
+
+          if (bodymatch) {
+            result.body_size = bodymatch[0].length;
+          }
+
+          let commentMatches = rawHtml.match(/<!--/g);
+
+          if (commentMatches) {
+            result.comment_count = commentMatches.length;
+          }
+
+          let ifCommentMatches = rawHtml.match(/<!-- *\[ *if/gsi);
+
+          if (ifCommentMatches) {
+            result.conditional_comment_count = ifCommentMatches.length;
+          }
+        }
+
+        return result;
+      }
+      catch(e) {
+        return logError("raw_html", e);
+      }
+    })(),
+    // canonicals used in the page and http header
+    // Used by: SEO
+    // Backup: link nodes in almanac.js
+    'canonicals': (() => {
+      try {
+        function processCanonical(c) {
+          if (c === result.url) 
+            result.self_canonical = true;
+        else 
```

Comment:
Weird whitespace

Reply:
as in it was not tabbed correctly? Fixed that.